packages/cumulus-api-client/tests/test-CumulusApiClient.js

```diff
@@ -232,3 +262,22 @@ test.serial('get retries/creates a token the expected number of times, then thro
     gotRevert();
   }
 });
+
+test.serial('_createAndUpdateNewAuthToken creates a new token and updates the cache', async (t) => {
+  const token = randomId();
+  const testApiClient = new CumulusApiClientRewire(t.context.config);
+  testApiClient._validateTokenExpiry = async () => true;
+  testApiClient.createNewAuthToken = async () => token;
+  await testApiClient._createAndUpdateNewAuthToken();
+  const actual = await testApiClient.getCacheAuthToken();
+  t.is(token, actual);
```

Comment:
I'm probably missing something - but how does this test verify that the cache itself is updated?

Reply:
You're not.   Two issues: it needs to be set to initialized on setup, and the test needs to have a new mock set for 'createNewAuthToken' to prove it's not generating the same thing.     

Nice catch, I'm glad you were taking a look at this. 

5e9aaa289

Reply:
Better refactored: d7b38ac51